### ISSUE TITLE
dependency alias is not working using pnpm

### ISSUE THREAD

**Issue by sxzz** [nuxt/framework#7590]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.17.0`
- Nuxt Version:     `3.0.0-rc.9`
- Nitro Version:    `0.5.1`
- Package Manager:  `npm@8.15.0`
- Builder:          `vite`
- User Config:      `meta`, `build`, `typescript`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

Minimum Reproduction: https://github.com/sxzz/nuxt-issue

Full Reproduction: https://github.com/element-plus/element-plus-nuxt-starter

### Describe the bug

When using pnpm (`shamefully-hoist`), dependency alias (`"@popperjs/core": "npm:@sxzz/popperjs-es@^2.11.7"`) is not working. But it works using npm.

```
Cannot find module '/path/nuxt-issue/.output/server/node_modules/@popperjs/core/dist/index.mjs' imported from /path/nuxt-issue/.output/server/node_modules/element-plus/es/components/popper/src/content.mjs
```

**pnpm:** Doesn't work. The directory of `.output/server/node_modules` is broken. There is only one file 
`package.json` in `.output/server/node_modules/@popperjs/core`. And `dist/index.mjs` goes to `.output/server/node_modules/@sxzz/popperjs-es/dist/index.mjs`

<img width="254" alt="image" src="https://user-images.githubusercontent.com/6481596/190619366-f7a8fcca-c8f2-43d7-b0a3-a2fb2dbea431.png">

---

**npm:** Works. `.output/server/node_modules/@sxzz/popperjs-es` is not existing. There are `package.json` and `dist` in `.output/server/node_modules/@popperjs/core`

<img width="255" alt="image" src="https://user-images.githubusercontent.com/6481596/190619042-83490ae1-1ca2-4702-8852-2037ee9893ee.png">

### Additional context

Relate #939

**Comment by danielroe** [comment]
This should be resolved in nitropack v0.5.4. which you can get with `yarn upgrade nuxt` or `npm upgrade nuxt`.

Let me know if not and I'll reopen.

**Comment by sxzz** [comment]
It works, thanks a lot!

**Comment by sxzz** [comment]
It works on my machine with node, but failed on vercel...

```
2022-09-22T23:20:43.708Z	199c1204-5981-4edc-bd71-2badf530965b	ERROR	[nuxt] [request error] [unhandled] [500] Cannot find package '@popperjs/core' imported from /var/task/node_modules/element-plus/es/components/popper/src/content.mjs
  at new NodeError (node:internal/errors:372:5)  
  at packageResolve (node:internal/modules/esm/resolve:954:9)  
  at moduleResolve (node:internal/modules/esm/resolve:1003:20)  
  at defaultResolve (node:internal/modules/esm/resolve:1218:11)  
  at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)  
  at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)  
  at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)  
  at link (node:internal/modules/esm/module_job:78:36)
```

**Comment by danielroe** [comment]
Hm. Any chance that Vercel might be using a different version of node or npm?

**Comment by sxzz** [comment]
Same node 16 and pnpm 7...

**Comment by danielroe** [comment]
Strange - so you are saying if I deploy your reproduction earlier to vercel, it will fail?

**Comment by sxzz** [comment]
Yes, you can try to deploy the [reproduction](https://github.com/sxzz/nuxt-issue) to vercel and check the errors.

Preview: https://nuxt-issue-q04o6xuwm-sxzz.vercel.app/

**Comment by sxzz** [comment]
I tried `NITRO_PRESET=vercel pnpm build` on my machine, and I found that there are only `@sxzz/popperjs-es`, and no `@popperjs/core`, just like `Describe the bug` in the issue body. I think that's why vercel serverless failed. 

Maybe the bug is fixed in node preset, and hasn't been fixed in vercel preset?
